databricks/koalas/tests/test_groupby.py

```diff
@@ -1161,6 +1161,16 @@ def test_apply_with_new_dataframe_without_shortcut(self):
         with option_context("compute.shortcut_limit", 0):
             self.test_apply_with_new_dataframe()
 
+    def test_apply_key_handling(self):
+        pdf = pd.DataFrame(
+            {"d": [1.0, 1.0, 1.0, 2.0, 2.0, 2.0], "v": [1.0, 2.0, 3.0, 4.0, 5.0, 6.0]}
+        )
+
+        self.assert_eq(pdf.groupby("d").sum(), pdf.groupby("d").apply(sum).drop(["d"], axis=1))
+        self.assert_eq(
+            ks.from_pandas(pdf).groupby("d").sum(), ks.from_pandas(pdf).groupby("d").apply(sum)
+        )
```

Comment:
Maybe

```python
        pdf = pd.DataFrame(
            {"d": [1.0, 1.0, 1.0, 2.0, 2.0, 2.0], "v": [1.0, 2.0, 3.0, 4.0, 5.0, 6.0]}
        )
        kdf = ks.from_pandas(pdf)

        self.assert_eq(pdf.groupby("d").apply(sum), kdf.groupby("d").apply(sum))
```

seems more clearer test for this bug fixing??

Reply:
Yes, that seems simpler. I could do

```python
        pdf = pd.DataFrame(
            {"d": [1.0, 1.0, 1.0, 2.0, 2.0, 2.0], "v": [1.0, 2.0, 3.0, 4.0, 5.0, 6.0]}
        )
        kdf = ks.from_pandas(pdf)

        self.assert_eq(pdf.groupby("d").apply(sum).drop(['d'], axis=1), kdf.groupby("d").apply(sum))
```

the `drop` is required since pandas will return d as a column and as key